web/packages/teleterm/src/ui/Search/SearchBar.tsx

```diff
@@ -83,9 +86,35 @@ function SearchBar() {
     }
   }, [close, isOpen, addWindowEventListener]);
 
-  function handleOnFocus(e: React.FocusEvent) {
-    open(e.relatedTarget);
-  }
+  // closeIfAnotherElementReceivedFocus handles a scenario where the focus shifts from the search
+  // input to another element on page. It does nothing if there's no other element that receives
+  // focus, i.e. the user clicks on an unfocusable element (for example, the empty space between the
+  // search bar and the profile selector).
+  //
+  // If that element is present though, onBlur takes precedence over onClickOutside. For example,
+  // clicking on a button outside of the search bar will trigger onBlur and will not trigger
+  // onClickOutside.
+  const closeIfAnotherElementReceivedFocus = makeEventListener(
```

Comment:
This is an unorthodox name, but I don't like listeners being called like `onBlur` and `handleBlur` if they do something more complex because then you have no idea what's happening in them.

That's the same reason why I renamed the prop in the following piece of code from `onShowDetails` to `showErrorsInModal`:

https://github.com/gravitational/teleport/blob/5bf72af28a5f3fdacd09717a6b5bad0a08b2eb60/web/packages/teleterm/src/ui/Search/pickers/ActionPicker.tsx#L217-L220

Reply:
Hmm maybe the complexity of the handler is the wrong way to look at it. What I should've said is that if I'm writing a generic component that's going to be used in many different contexts, say a `<Button>` or `<Input>`, then `onClick`, `onBlur` accepted through props all make sense because `<Button>` doesn't care what the handler does.

But if I'm working on a specific component in a specific context, then IMHO giving specific names to handlers makes it easier to understand what's going on, as you described. Otherwise all I see is just `onBlur` or `handleBlur` and if I want to know what actually happens on blur, I have to go to the function _and_ I have to read through it because there's no variable name which would summarize what the function does.

Matter of fact, the `onInputValueChange` function exported from `SearchContext` has this problem too, I'm going to rename it in the next commit.

Reply:
I think it makes sense to use more specific names when possible. 
If I needed to run a few unrelated actions in a handler, then probably I would just call it `handleBlur`.